app/controllers/sessions_controller.rb

```diff
@@ -202,7 +202,11 @@ def destroy
       url ||= begin
         referrer_uri = URI(request.referer)
         request_uri = URI(request.url)
-        "#{referrer_uri.scheme}://#{referrer_uri.host}:#{referrer_uri.port}/?#{request_uri.query}"
+        if referrer_uri.host == request_uri.host
+          "#{root_url}?#{request_uri.query}"
+        else
+          "#{referrer_uri.scheme}://#{referrer_uri.host}:#{referrer_uri.port}/?#{request_uri.query}"
+        end
```

Comment:
if Accounts is the referrer, we want to use `root_url` so that we automatically get the path prefixed URL.

Reply:
hm, this is kinda dangerous.  `referer` is getting more and more unsupported due to browser security restrictions.  https://stackoverflow.com/questions/6880659/in-what-cases-will-http-referer-be-empty

It does seem like it should be set if it's coming from the same site, but don't know if we can rely on that in all cases.  

However I can't think of any other method to use here 🤔

Reply:
I agree this isn't the best, but at least this PR doesn't change the referrer here (that line was already here), it just adds a new case for when Accounts is the referrer